### PR TITLE
Add 1 to device_id length to accommodate trailing null.

### PR DESCRIPTION
**Pull request by patrickvacek**
Fixing a bug caught by @rdanitz

### CODE REVIEW

**Comment by cajun-rat** [comment]
Ouch.  Is this bug anywhere else in the codebase?

**Comment by patrickvacek** [comment]
I hope not. This was from the new code that reads the device_id from the common name of the certificate. To be fair, the documentation is rather vague about whether space for the trailing null is necessary.

File: src/uptane/initialize.cc [hunk 1]
```diff
@@ -38,8 +38,8 @@ bool Repository::initDeviceId(const ProvisionConfig& provision_config, const Upt
         X509_free(x);
         return false;
       }
-      boost::scoped_array<char> buf(new char[len]);
-      X509_NAME_get_text_by_NID(X509_get_subject_name(x), NID_commonName, buf.get(), len);
+      boost::scoped_array<char> buf(new char[len + 1]);
+      X509_NAME_get_text_by_NID(X509_get_subject_name(x), NID_commonName, buf.get(), len + 1);
```
Comment (OYTIS) [inline src/uptane/initialize.cc]: I'm not sure if we need 'len+1' here though. The documentation is a bit ambiguous on this account:

> At most len bytes will be written and the text written to buf will be null terminated

https://www.openssl.org/docs/man1.1.0/crypto/X509_NAME_get_text_by_NID.html

Comment (OYTIS) [reply src/uptane/initialize.cc]: I mean the last argument to the second call to X509_NAME_get_text_by_NID

Comment (patrickvacek) [reply src/uptane/initialize.cc]: I had the same concern, so I tested both ways. It is required.

**Pull request merged**